### PR TITLE
Adrien/deal for ob

### PR DESCRIPTION
**Pull request by adhusson**
the main purpose is that the custom orderbook can now deal with arbitrary tokens, on forks (and not, as before, only on tokens with permisionless minting).

ready for review 
has been decomposed into meaningful commits so review may be easier commit-by-commit

### CODE REVIEW

File: packages/mangrove.js/src/cli/commands/dealCmd.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    {
+      spawn: false,
+      deploy: false,
+      host: argv.host,
+      port: argv.port,
+      pipe: true,
+    },
+    false
+  ).connect();
+  console.log("hey?");
```
Comment (peterMangrove) [inline packages/mangrove.js/src/cli/commands/dealCmd.ts:27]: Should this be removed

File: packages/mangrove.js/src/script.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import { Mangrove } from "./index";
+console.log("hellO?");
```
Comment (peterMangrove) [inline packages/mangrove.js/src/script.ts:2]: Should this be removed?

**Review by peterMangrove — CHANGES_REQUESTED** [review]

File: packages/mangrove.js/src/cli/commands/dealCmd.ts (hunk 1 above)
Comment (adhusson) [reply packages/mangrove.js/src/cli/commands/dealCmd.ts:27]: yes :D

File: packages/mangrove.js/src/script.ts (hunk 2 above)
Comment (adhusson) [reply packages/mangrove.js/src/script.ts:2]: as well!

Comment (adhusson) [reply packages/mangrove.js/src/script.ts:2]: the whole file actually

**Pull request merged**